firewood/src/proof.rs

```diff
@@ -148,23 +148,30 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
         mut key_nibbles: NibblesIterator<'a, 0>,
         encoded_node: &[u8],
     ) -> Result<(Option<SubProof>, NibblesIterator<'a, 0>), ProofError> {
-        let items: Vec<Encoded<Vec<u8>>> = bincode::DefaultOptions::new()
-            .deserialize(encoded_node)
-            .map_err(ProofError::DecodeError)?;
-
-        match items.len() {
-            EXT_NODE_SIZE => {
-                let mut items = items.into_iter();
-                let decoded_key: Vec<u8> = items.next().unwrap().decode()?;
+        let node = NodeType::decode(encoded_node)?;
```

Comment:
Remove the nesting! `locate_subproof` is only called in one place. Change the signature as follows:
```rust
    fn locate_subproof<'a>(
        &self,
        mut key_nibbles: NibblesIterator<'a, 0>,
        node: NodeType,
    ) -> Result<(Option<SubProof>, NibblesIterator<'a, 0>), ProofError> {
```